File: python/avoid_flag_arguments.py
# Avoid Flag Arguments
# Avoid providing boolean arguments (true/false) to functions or methods.
# You could pass a string with a clearer name (for example).

# Before
def greet(name, formal):
    if formal:
        print(f"Good evening, {name}.")
    else:
        print(f"Hi, {name}!")

greet("Alice", True)
greet("Bob", False)


# After
def greet(name, formality):
    if formality == "formal":
        print(f"Good evening, {name}.")
    elif formality == "informal":
        print(f"Hi, {name}!")

greet("Alice", "formal")
greet("Bob", "informal")

File: kotlin/avoid_flag_arguments.kt
// Avoid Flag Arguments
// Avoid providing boolean arguments (true/false) to functions or methods.
// You could pass a string with a clearer name (for example).

// Before
fun greetBefore(name: String, formal: Boolean) {
    if (formal) {
        println("Good evening, $name.")
    } else {
        println("Hi, $name!")
    }
}

// After
fun greet(name: String, formality: String) {
    when (formality) {
        "formal" -> println("Good evening, $name.")
        "informal" -> println("Hi, $name!")
    }
}

fun main() {
    greetBefore("Alice", true)
    greetBefore("Bob", false)

    println()

    greet("Alice", "formal")
    greet("Bob", "informal")
}
